### PR TITLE
Add missing user-facing part to allow volume subpath mounts

### PR DESCRIPTION
**Pull request by basilgello**
Add missing user-facing "subpath" option to mount subpaths of named volumes.

#### Does this PR introduce a user-facing change?

Yes, adding "subpath" as valid mount and volume option in podman-container-create(1), podman-run(1).

```release-note
Allow users to mount subpaths of named volumes via 'subpath' mount option.
```

### CODE REVIEW

**Comment by basilgello** [comment]
After #17992 was merged, I had to add only user-facing change to fix #20661

This is a squashed commit with vendors in for easier review, I will split the relevant commits into proper PRs once I get general ACK.

File: pkg/specgenutil/volumes.go [hunk 1]
```diff
@@ -371,6 +371,19 @@ func parseMountOptions(mountType string, args []string) (*spec.Mount, error) {
 				return nil, fmt.Errorf("host directory cannot be empty: %w", errOptionArg)
 			}
 			mnt.Source = value
+		case "subpath":
+			if setSubpath {
+				return nil, fmt.Errorf("cannot pass 'subpath' option more than once: %w", errOptionArg)
```
Comment (rhatdan) [inline pkg/specgenutil/volumes.go:376]: nit: This error should use %q and name like the one just below.

**Comment by basilgello** [comment]
@giuseppe you mentioned the security implications of this features in the linked issue. I think 17992 solved them, so please review.

File: vendor/github.com/containers/common/pkg/parse/parse.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ import (
 
 // ValidateVolumeOpts validates a volume's options
 func ValidateVolumeOpts(options []string) ([]string, error) {
-	var foundRootPropagation, foundRWRO, foundLabelChange, bindType, foundExec, foundDev, foundSuid, foundChown, foundUpperDir, foundWorkDir, foundCopy, foundCopySymlink int
+	var foundRootPropagation, foundRWRO, foundLabelChange, bindType, foundExec, foundDev, foundSuid, foundChown, foundUpperDir, foundWorkDir, foundCopy, foundCopySymlink, foundVolumeSubpath int
```
Comment (giuseppe) [inline vendor/github.com/containers/common/pkg/parse/parse.go:17]: these changes must go in the https://github.com/containers/common project and then revendor them

**Comment by giuseppe** [comment]
> @giuseppe you mentioned the security implications of this features in the linked issue. I think 17992 solved them, so please review.

yes, these are solved with #17992

File: vendor/github.com/containers/common/pkg/parse/parse.go (hunk 2 above)
Comment (basilgello) [reply vendor/github.com/containers/common/pkg/parse/parse.go:17]: I understand that, but I need some details.

Is the correct procedure here as follows?

* Move this commit into separate PR in containers/common
* Wait for it to be merged
* `go get -u github.com/containers/common && go mod vendor` in this PR
* Rebase this PR with new `go.{mod,sum}` and `vendor`

Comment (basilgello) [reply vendor/github.com/containers/common/pkg/parse/parse.go:17]: @giuseppe bump?

Comment (giuseppe) [reply vendor/github.com/containers/common/pkg/parse/parse.go:17]: sorry, forgot to submit the review.

**Comment by larsks** [comment]
Docker has implemented this feature using the `volume-subpath` mount option (see [pr 45687](https://github.com/moby/moby/pull/45687), landed on 19-jan-2024). This is available now in Docker v26.0.0.

It's probably worth updating this pr to adopt the same option names so that the cli and api are compatible.

**Comment by mheon** [comment]
@basilgello Poke - are you still working on this? If not, we may take over to push this over the finish line

**Comment by basilgello** [comment]
Let me spliy the PR tomorrow morning @mheon Should we follow up Docker convention as larsks mentioned or the current syntax is fine?

**Comment by basilgello** [comment]
Also I postponded the split because I expected to get primary ACK from @giuseppe or other code owner but I did not get it to the date + several other issues on my side.

**Comment by mheon** [comment]
Docker convention would be preferred.

Sorry for the sudden attention, we're starting to prep for Podman 5.1 and this came up as something that could get in.

**Comment by basilgello** [comment]
It is OK and I needed that nudge :) If apart from the Docker convention the rest of PR is OK, I will do the split and link the derived PRs to this one.

**Comment by basilgello** [comment]
@mheon I need your help :) In #22410 you added `subpath` option to image mounts. But Docker chose to name subpath option as `volume-subpath` (as larsks mentioned, just the [wrong PR](https://github.com/docker/cli/pull/4331)) Now if I change the user-facing option name for volume and mount to Docker's one, should I also rename the image mount option and its docs?

AFAIK the merged code is still in dev branch so I think it is better to rename all user-facing `subpath` option name to `volume-subpath`. WDYT?

**Comment by basilgello** [comment]
@mheon Check your corp mailbox please!

**Comment by JCBird1012** [comment]
Is this still necessary after #24532?

**Pull request closed without merging**